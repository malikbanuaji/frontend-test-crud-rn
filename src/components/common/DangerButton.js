import React from 'react';
import {StyleSheet} from 'react-native';
import {Mixins, Typography} from '../../styles';
import BaseButton from './BaseButton';

export default function DangerButton({text, onPress, textStyle, style}) {
  return (
    <BaseButton
      text={text}
      onPress={onPress}
      style={[styles.button, style]}
      textStyle={[styles.buttonText, textStyle]}
    />
  );
}

const styles = StyleSheet.create({
  button: {
    ...Mixins.buttonDanger,
  },
  buttonText: {
    ...Typography.buttonDangerText,
  },
});
