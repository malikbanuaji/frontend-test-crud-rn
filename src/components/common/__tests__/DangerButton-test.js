import React from 'react';
import renderer from 'react-test-renderer';
import DangerButton from '../DangerButton';

test('renders correctly', () => {
  const tree = renderer.create(<DangerButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
