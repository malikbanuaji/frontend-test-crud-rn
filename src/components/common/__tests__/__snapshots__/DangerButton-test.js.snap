// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#CA0B00",
          "borderRadius": 4,
          "overflow": "hidden",
        },
        undefined,
      ],
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Rubik-Medium",
            "fontSize": 16,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Rubik-Medium",
              "fontSize": 16,
              "textAlign": "center",
            },
            undefined,
          ],
        ]
      }
    />
  </View>
</View>
`;
